src/ZoweUSSNode.ts

```diff
@@ -278,14 +278,21 @@ export class ZoweUSSNode extends ZoweTreeNode implements IZoweUSSTreeNode {
      * @param oldReference string
      * @param revision string
      */
-    public rename(newFullPath: string) {
+    public async rename(newFullPath: string) {
         this.fullPath = newFullPath;
         this.shortLabel = newFullPath.split("/").pop();
         this.label = this.shortLabel;
         this.tooltip = injectAdditionalDataToTooltip(this, newFullPath);
 
-        vscode.commands.executeCommand("zowe.uss.refreshUSSInTree", this);
-        vscode.commands.executeCommand("zowe.uss.ZoweUSSNode.open", this);
+        if (this.isFolder) {
+            await vscode.commands.executeCommand("zowe.uss.refreshAll");
+        } else {
+            await vscode.commands.executeCommand("zowe.uss.refreshUSSInTree", this);
+        }
+
+        if (!this.isFolder) {
+            await vscode.commands.executeCommand("zowe.uss.ZoweUSSNode.open", this);
```

Comment:
Does this line need to be in its own `if (!this.isFolder)` condition, or could it be part of the `else` block at line 289, after the `refreshUSSInTree` command?

Reply:
I did it in this form to align with more advanced implementation in another PR.
https://github.com/zowe/vscode-extension-for-zowe/blob/af3719e4aff62fb7bd0fa470a0b3ca6470b1d35a/src/ZoweUSSNode.ts#L292

If it's a problem I can change it - no worries

Reply:
@lauren-li Please, let me know if you have other concerns

Reply:
@stepanzharychevbroadcom To me, it makes more sense to have this line as part of the `else` block for now, and then later move it into its own `if (!this.isFolder) &&...` block in the PR for the more advanced implementation.

Reply:
@lauren-li Changed it.

Reply:
Thanks @stepanzharychevbroadcom !